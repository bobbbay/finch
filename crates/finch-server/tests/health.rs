//! A general health check.

use axum::{
    body::Body,
    http::{Request, StatusCode},
};
use finch_server::{app, error::Result};
use tower::ServiceExt;

#[tokio::test]
async fn health() -> Result<()> {
    let app = app().await?;

    let response = app
        .oneshot(
            Request::builder()
                .uri("/health")
                .body(Body::empty())
                .unwrap(),
        )
        .await
        .unwrap();

    assert_eq!(response.status(), StatusCode::OK);

    let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
    insta::assert_debug_snapshot!(body);

    Ok(())
}
