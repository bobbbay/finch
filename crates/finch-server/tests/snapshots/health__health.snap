---
source: crates/finch-server/tests/health.rs
expression: body
---
b""
